### PR TITLE
[ticket/16902] Improve search results count - 3.3.x

### PR DESCRIPTION
**Pull request by rxu**
`3.3.x` version.

Checklist:

- [x] Correct branch: master for new features; 3.3.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/master/coding-guidelines.html) and [3.3.x](https://area51.phpbb.com/docs/dev/3.3.x/development/coding_guidelines.html)
- [x] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.3.x/development/git.html)

<a href="https://tracker.phpbb.com/browse/PHPBB3-16902">PHPBB3-16902</a>.

### CODE REVIEW

File: phpBB/phpbb/search/fulltext_mysql.php [hunk 1]
```diff
@@ -828,12 +827,12 @@ public function author_search($type, $firstpost_only, $sort_by_sql, $sort_key, $
 		extract($this->phpbb_dispatcher->trigger_event('core.search_mysql_author_query_before', compact($vars)));
 
 		// If the cache was completely empty count the results
-		$calc_results = ($result_count) ? '' : 'SQL_CALC_FOUND_ROWS ';
+		$sql_select	= ($type == 'posts') ? 'p.post_id' : 't.topic_id';
```
Comment (Nicofuma) [inline phpBB/phpbb/search/fulltext_mysql.php:830]: you could remove this variable to simplify the code (like before)

Comment (rxu) [reply phpBB/phpbb/search/fulltext_mysql.php:830]: Well, it's used later for wrapping into `COUNT()`, so if I remove it here that would be the trick.

File: tests/test_framework/phpbb_functional_test_case.php [hunk 2]
```diff
@@ -1433,4 +1433,112 @@ protected function get_hidden_fields($crawler, $url)
...
+	/**
+	 * Get HTML of the crawler
+	 * See https://symfony.com/doc/current/components/dom_crawler.html#component-dom-crawler-dumping
+	 *
+	 * @param Symfony\Component\DomCrawler\Crawler $crawler Crawler instance
+	 * @param string $url Request URL
+	 *
+	 * @return array Hidden form fields array
+	 */
+	protected function dump_crawler($crawler)
```
Comment (marc1706) [inline tests/test_framework/phpbb_functional_test_case.php:1446]: Is this needed?

File: tests/test_framework/phpbb_functional_test_case.php [hunk 3]
```diff
@@ -1433,4 +1433,112 @@ protected function get_hidden_fields($crawler, $url)
...
+		{
+			$username_logged_in = $crawler->filter('li[id="username_logged_in"] > div > a > span')->text();
+		}
+		return $username_logged_in;
+	}
+
+	/**
+	 * Disable posting flood control
+	 */
+	protected function disable_flood_interval()
```
Comment (marc1706) [inline tests/test_framework/phpbb_functional_test_case.php:1482]: Currently this is only used in one place. Maybe replace it with a static function that can be called from other tests cases if needed at some point? Also, could you also add a function to enable the flood interval again? Ideally we should aim to try and keep side effects of tests to a minimum.

**Review by marc1706 — CHANGES_REQUESTED** [review]

File: tests/test_framework/phpbb_functional_test_case.php (hunk 2 above)
Comment (rxu) [reply tests/test_framework/phpbb_functional_test_case.php]: Was handy for debugging tests but removed.

File: tests/test_framework/phpbb_functional_test_case.php (hunk 3 above)
Comment (rxu) [reply tests/test_framework/phpbb_functional_test_case.php]: Not sure about static but fixed as I could.

File: tests/test_framework/phpbb_functional_test_case.php [hunk 4]
```diff
@@ -1437,4 +1437,87 @@ protected function get_hidden_fields($crawler, $url)
...
+		{
+			$this->logout();
+			$this->login($logged_in_username);
+		}
+	}
+
+	/**
+	* Check if a user exists by username(s) or user_id(s)
+	*
+	* @param array &$user_id_ary The user ids to check or empty if usernames used
```
Comment (marc1706) [inline tests/test_framework/phpbb_functional_test_case.php:1501]: Param descriptions are not in line with function

**Review by marc1706 — CHANGES_REQUESTED** [review]
Apart from minor issue with docblocks it looks good.

File: tests/test_framework/phpbb_functional_test_case.php [hunk 5]
```diff
@@ -1496,10 +1496,10 @@ protected function set_flood_interval($flood_interval)
 	}
 
 	/**
-	* Check if a user exists by username(s) or user_id(s)
+	* Check if a user exists by username or user_id
 	*
-	* @param array &$user_id_ary The user ids to check or empty if usernames used
-	* @param array &$username_ary The usernames to check or empty if user ids used
+	* @param string $username The username to check or empty if user_id used
+	* @param int $user_id The use id to check or empty if username used
```
Comment (3D-I) [inline tests/test_framework/phpbb_functional_test_case.php:1502]: `The user id to check or empty if username used` ?

Comment (marc1706) [reply tests/test_framework/phpbb_functional_test_case.php:1502]: id instead of identifier is the way to go, should say user and not "use" though :)

**Review by marc1706 — APPROVED** [review]

**Pull request merged**